contracts/Lockbox.cdc

```diff
@@ -0,0 +1,329 @@
+
+    /// path to store the private locked token admin link
+    /// in the shared account
+    pub let LockedTokenAdminPrivatePath: Path
+
+    /// path to store the admin collection 
+    /// in the admin account
+    pub let LockedTokenAdminCollectionPath: Path
+
+    /// path to store the token holder resource
+    /// in the unlocked account
+    pub let TokenHolderStoragePath: Path
+
+    /// path to store the node operator struct
+    /// in the node operators account for staking helper
+    pub let NodeOperatorCapabilityStoragePath: Path
+
+    /// stored in Admin account
+    pub resource interface TokenAdmin {
+        pub fun increaseUnlockLimit(delta: UFix64)
+    }
+
+    /// stored in the shared account 
+    pub resource LockedTokenManager: FungibleToken.Receiver, FungibleToken.Provider, TokenAdmin {
+    
+        pub var vault: Capability<&FlowToken.Vault>
+
+        pub var nodeStaker: @FlowIDTableStaking.NodeStaker?
+
+        pub var nodeDelegator: @FlowIDTableStaking.NodeDelegator?
+
+        pub var unlockLimit: UFix64
+
+        init(vault: Capability<&FungibleToken.Vault>) {
+            self.vault = vault
+            self.nodeStaker = nil
+            self.nodeDelegator = nil
+            self.unlockLimit = 0.0
+        }
+
+        // FungibleToken.Receiver actions
+
+        pub fun deposit(from: @FungibleToken.Vault) {
+            let balance = from.balance
+
+            self.vault.deposit(from: <- from)
+
+            self.increaseUnlockLimit(delta: balance)
+        }
+
+        // FungibleToken.Provider actions
+
+        pub fun withdraw(amount: UFix64): @FungibleToken.Vault {
+            pre {
+                self.unlockLimit >= amount: "Requested amount exceeds unlocked token limit"
+            }
+
+            post {
+                self.unlockLimit == before(self.unlockLimit) - amount: "Updated unlocked token limit is incorrect"
+            }
+
+            let vault <- self.vault.withdraw(amount: amount)
+
+            self.decreaseUnlockLimit(delta: amount)
+
+            return <- vault
+        }
+
+        access(self) fun decreaseUnlockLimit(delta: UFix64) {  
+            self.unlockLimit = self.unlockLimit - delta
+        }
+
+        // Lockbox.TokenAdmin actions
+
+        pub fun increaseUnlockLimit(delta: UFix64) {  
+            self.unlockLimit = self.unlockLimit + delta
+        }
+
+        // Lockbox.TokenHolder actions
+
+        pub fun registerNode(id: String, role: UInt8, networkingAddr: String, networkingKey: String, stakingKey: String) {
+            self.nodeStaker <- FlowIDTableStaking.addNodeRecord(self.nodeInfo)
+        }
+
+        pub fun registerDelegator(nodeID: String) {
+            self.nodeDelegator <- FlowIDTableStaking.registerDelegator(nodeID: nodeID)
+        }
+
+        pub fun createTokenHolder(tokenManager: Capability<&LockedTokenManager>): @TokenHolder {
+            return <- create TokenHolder(tokenManager: tokenManager)
+        }
```

Comment:
Can this just be defined directly on the `Lockbox` contract?

Reply:
yeah I guess it could. I'll do that